### ISSUE TITLE
Add support for basic auth (uptimerobot.com)

### ISSUE THREAD

**Issue by xanonid** [stakater/IngressMonitorController#148]
uptimerobot.com supports adding basic auth (http_username & http_password) to monitors. It would be nice, if IngressMonitorController could set those settings via related Ingress annotations/secrets.

**Comment by SinFulNard** [comment]
This also applies to Pingdom!

**Comment by SinFulNard** [comment]
I've submitted a PR to bring this to Pingdom via an annotation and OS env (mounted secret); https://github.com/stakater/IngressMonitorController/pull/154

**Comment by DhiaTN** [comment]
I've submitted a PR implementation for Statuscake  #170

**Comment by grebois** [comment]
@xanonid @SinFulNard @DhiaTN why not having an annotation pointing to a secret for the credentials? this way you dont need to mount the secret into the deployment/pod for the get.env

**Comment by SinFulNard** [comment]
@grebois - I didn't have the thought at the time (or experience to achieve that). So went with what I could think of. 

I agree, its likely a better method of exposing the secret.

**Comment by grebois** [comment]
@SinFulNard are you at kubecon this next week? if you are lets talk about it f2f

**Comment by hazim1093** [comment]
Hi @grebois, me and @waseem-h will be there at KubeCon. We'd be happy to meet if you want to discuss something or just say Hi!

**Comment by grebois** [comment]
Definitely, let's catch up, are you already in Barcelona? whazatpp me +49 152 0249 6483

**Comment by grebois** [comment]
@hazim1093 @waseem-h @SinFulNard please leave this open for a bit longer I will work on a PR together with @DhiaTN to refactor this functionality, including https://github.com/stakater/IngressMonitorController/issues/182 and https://github.com/stakater/IngressMonitorController/issues/183

**Comment by hazim1093** [comment]
> Definitely, let's catch up, are you already in Barcelona? whazatpp me +49 152 0249 6483

Hi, Sorry I didn't get notification for the comment, just saw :(

**Comment by rasheedamir** [comment]
@grebois will you continue on this one?

**Comment by grebois** [comment]
@rasheedamir not for the moment, but I would like to keep it open.

**Comment by rasheedamir** [comment]
ok sure! @grebois

**Comment by karl-johan-grahn** [comment]
Reopening issues that inadvertently were closed as stale
